rank/src/services/order.ts

```diff
@@ -0,0 +1,79 @@
+import { TestCase, TestResult } from '../types/test'
+
+import { getClient } from './test'
+import { getTemplate } from './search-templates'
+import tests from '../tests'
+
+function checkResponseOrder(searchResponse, testCase: TestCase): boolean {
+  const beforeHits = searchResponse.hits.hits.filter((hit) =>
+    testCase.before.includes(hit._id)
+  )
+
+  const afterHits = searchResponse.hits.hits.filter((hit) =>
+    testCase.after.includes(hit._id)
+  )
+
+  const pass = beforeHits.every((beforeHit) => {
+    return afterHits.every((afterHit) => {
+      return beforeHit._score > afterHit._score
+    })
+  })
+
+  return pass
+}
+
+async function testOrder({
+  queryEnv,
+  cluster,
+  index,
+  testId
+}): Promise<TestResult> {
+  // get the test data
+  const template = await getTemplate(queryEnv, index)
+
+  const test = tests[template.namespace].find((test) => test.id === testId)
+  if (!test) throw Error(`No such test ${testId}`)
+
+  const results = await Promise.all(
+    // for each test case, run a search
+    test.cases.map(async (testCase) => {
+      const { searchTerms, description, knownFailure } = testCase
+      const client = await getClient(queryEnv, cluster, template)
```

Comment:
Can you move this out to the top-level, reuse the same client rather than fetching it each time?